### ISSUE TITLE
v0.4.5

### ISSUE THREAD

**Issue by ThomasBlock** [swanchain/go-computing-provider#40]
### Ubi

no longer appearing ( it's 15:00:00 )

```
36979  	CPU      	fil-c2-512M	                                                                  	failed 	0.0   	2024-03-09 03:25:32	
37021  	CPU      	fil-c2-512M	                                                                  	failed 	0.0   	2024-03-09 03:55:31	
37064  	CPU      	fil-c2-512M	                                                                  	failed 	0.0   	2024-03-09 04:25:32	
37106  	CPU      	fil-c2-512M	                                                                  	failed 	0.0   	2024-03-09 04:55:32	
37149  	CPU      	fil-c2-512M	                                                                  	failed 	0.0   	2024-03-09 05:25:32	

```
### Init

```
computing-provider init --ownerAddress 0xfe017Ff8F0C7349845Ab52E58FcA96143f2c4981 --beneficiaryAddress 0x269EBeee083CE6f70486a67dC8036A889bF322A9
Contract deployed! Address: 0x8A46F39BfDFA7edc9538F27bB6D2E78597b62C0b
Transaction hash: 0x69c51fe629f7b07a5f8c374a296696079e38cbf0ab9bca97d4c8adc2d3715182
time="2024-03-09 12:05:24.717" level=error msg="response convert to json failed: unexpected end of JSON input" func=DoSend file="run.go:699"
Error: unexpected end of JSON input

```

### Collateral

collateral still inaccurate

HUB info for beneficiary: 8.65 sETH
HUB info for owner: 1.49 sETH
computing-provider info: Collateral(SWAN-ETH): 0.95000

### renting

renting still problem.

case A)

```
[Registry]
ServerAddress = ""      
```

lagrange GPU renting with stablediffusion is initalizing with all its logs

```
{"stream":" ---\u003e Running in 67873a2ed19f\n"}
{"stream":" ---\u003e dff85580116d\n"}
{"aux":{"ID":"sha256:dff85580116d0071f9a552a900e68846bf2c39729516a31bf1c579599f67298f"}}
{"stream":"Successfully built dff85580116d\n"}
{"stream":"Successfully tagged lagrange/stable-diffusion-bse-lora-df49358c96ee:1709982502\n"}
time="2024-03-09 12:10:39.340" level=info msg="Start deleting space service, space_uuid: 671cd2fb-4e80-4107-b089-df49358c96ee" func=deleteJob file="cp_service.go:1095"
```
but then we can not pull image 
` Failed to pull image`

`deploy-671cd2fb-4e80-4107-b089-df49358c96ee-7d6ccdd445-j7rhb   0/1     ErrImagePull       0                  14s`

```
kubectl describe pod 
Events:
  Type     Reason     Age                From               Message
  ----     ------     ----               ----               -------
  Normal   Scheduled  40s                default-scheduler  Successfully assigned ns-../deploy-671cd2fb-4e80-4107-b089-df49358c96ee-7d6ccdd445-j7rhb to swan3
  Normal   Pulling    25s (x2 over 41s)  kubelet            Pulling image "lagrange/stable-diffusion-bse-lora-df49358c96ee:1709982502"
  Warning  Failed     25s (x2 over 40s)  kubelet            Failed to pull image "lagrange/stable-diffusion-bse-lora-df49358c96ee:1709982502": failed to pull and unpack image "docker.io/lagrange/stable-diffusion-bse-lora-df49358c96ee:1709982502": failed to resolve reference "docker.io/lagrange/stable-diffusion-bse-lora-df49358c96ee:1709982502": pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
  Warning  Failed     25s (x2 over 40s)  kubelet            Error: ErrImagePull
  Normal   BackOff    10s (x2 over 39s)  kubelet            Back-off pulling image "lagrange/stable-diffusion-bse-lora-df49358c96ee:1709982502"
  Warning  Failed     10s (x2 over 39s)  kubelet            Error: ImagePullBackOff
```

case B)

```
[Registry]
ServerAddress = "http://192.168.128.71:5000"    
```

```
time="2024-03-09 14:55:04.364" level=info msg="Job received Data: {UUID:fea9f7cb-d5ea-492f-9b0c-129d5c80fa4b Name:Job-fea9f7cb-d5ea-492f-9b0c-129d5c80fa4b Status:Submitted Duration:3600 JobSourceURI:https://api.lagrangedao.org/spaces/201d7532-8284-4ebd-b348-ff5ce862beda JobResultURI: StorageSource:lagrange TaskUUID:f379bb27-ad8a-4099-bb4c-daeab0590156 CreatedAt:1709992504 UpdatedAt:1709992504 BuildLog: ContainerLog: NodeIdJobSourceUriSignature:0x0bca7127734cdc9c2ab2630742d7388df65b896abe5a4da4c5d8ea316234d0bb56547d7a11c627428f5afbb4205c06cfabc45acb5b80c8279ae342775723eb7b1b}" func=ReceiveJob file="cp_service.go:79"
time="2024-03-09 14:55:04.892" level=info msg="checkResourceAvailableForSpace: needCpu: 8, needMemory: 16.00, needStorage: 30.00" func=checkResourceAvailableForSpace file="cp_service.go:1242"
time="2024-03-09 14:55:04.892" level=info msg="checkResourceAvailableForSpace: remainingCpu: 2, remainingMemory: 11.00, remainingStorage: 348.00" func=checkResourceAvailableForSpace file="cp_service.go:1243"
time="2024-03-09 14:55:04.892" level=info msg="checkResourceAvailableForSpace: needCpu: 8, needMemory: 16.00, needStorage: 30.00" func=checkResourceAvailableForSpace file="cp_service.go:1242"
time="2024-03-09 14:55:04.892" level=info msg="checkResourceAvailableForSpace: remainingCpu: 11, remainingMemory: 53.00, remainingStorage: 1573.00" func=checkResourceAvailableForSpace file="cp_service.go:1243"
time="2024-03-09 14:55:04.892" level=info msg="gpuName: NVIDIA-4090, nodeGpu: map[:0 kubernetes.io/os:0], nodeGpuSummary: map[swan3:map[NVIDIA-4090:1] swan7:map[NVIDIA-3090:1]]" func=checkResourceAvailableForSpace file="cp_service.go:1250"
time="2024-03-09 14:55:04.893" level=info msg="submitting job..." func=submitJob file="cp_service.go:149"
time="2024-03-09 14:55:05.144" level=info msg="uuid: 201d7532-8284-4ebd-b348-ff5ce862beda, spaceName: myDiffusion, hardwareName: Nvidia 4090 · 8 vCPU · 16 GiB" func=DeploySpaceTask file="cp_service.go:1051"
time="2024-03-09 14:55:05.172" level=info msg="uploading file to bucket, objectName: jobs/f2262edf-2c9b-4b35-92dc-743e98026a32.json, filePath: /tmp/jobs/f2262edf-2c9b-4b35-92dc-743e98026a32.json" func=UploadFileToBucket file="storage_service.go:52"
2024/03/09 14:55:05 Image path: build/0x7B0CEe1939a4AdA062EC79f4862a42C1F47B1806/spaces/myDiffusion
time="2024-03-09 14:55:05.657" level=error msg="Error building Docker image: Error response from daemon: invalid reference format" func=BuildImagesByDockerfile file="buildspace.go:79"
time="2024-03-09 14:55:05.657" level=info msg="Failed to extract exposed port: unable to open Dockerfile: open : no such file or directory" func=DockerfileToK8s file="deploy.go:97"
time="2024-03-09 14:55:06.337" level=error msg="http status: 400 Bad Request, code:400, url:https://api.swanipfs.com/api/v2/oss_file/get_file_by_object_name?bucket_uid=878494a8-6ab7-4694-96ac-fc89a2afcbe1&object_name=jobs/f2262edf-2c9b-4b35-92dc-743e98026a32.json" func=HttpRequest file="restful.go:127"
time="2024-03-09 14:55:06.337" level=error msg="https://api.swanipfs.com/api/v2/oss_file/get_file_by_object_name?bucket_uid=878494a8-6ab7-4694-96ac-fc89a2afcbe1&object_name=jobs/f2262edf-2c9b-4b35-92dc-743e98026a32.json failed, status:error, message:invalid param value:record not found" func=HttpRequest file="restful.go:154"
time="2024-03-09 14:55:06.338" level=error msg="https://api.swanipfs.com/api/v2/oss_file/get_file_by_object_name?bucket_uid=878494a8-6ab7-4694-96ac-fc89a2afcbe1&object_name=jobs/f2262edf-2c9b-4b35-92dc-743e98026a32.json failed, status:error, message:invalid param value:record not found" func=HttpGet file="restful.go:64"
time="2024-03-09 14:55:08.857" level=info msg="file name:1_f2262edf-2c9b-4b35-92dc-743e98026a32.json, chunk size:882" func=func1 file="file.go:248"
time="2024-03-09 14:55:11.248" level=info msg="jobuuid: fea9f7cb-d5ea-492f-9b0c-129d5c80fa4b successfully submitted to IPFS" func=submitJob file="cp_service.go:177"
time="2024-03-09 14:55:11.512" level=info msg="submit job detail: {UUID:fea9f7cb-d5ea-492f-9b0c-129d5c80fa4b Name:Job-fea9f7cb-d5ea-492f-9b0c-129d5c80fa4b Status:submitted Duration:3600 JobSourceURI:https://api.lagrangedao.org/spaces/201d7532-8284-4ebd-b348-ff5ce862beda JobResultURI:https://7d67303d2964.acl.swanipfs.com/ipfs/QmWjpH5nGGrfUxk7QxA13eqpWAxtnPsrYaQHx3DPmwLAX9 StorageSource:lagrange TaskUUID:f379bb27-ad8a-4099-bb4c-daeab0590156 CreatedAt:1709992504 UpdatedAt:1709992504 BuildLog:wss://log.bitstakehaven.com:8085/api/v1/computing/lagrange/spaces/log?space_id=201d7532-8284-4ebd-b348-ff5ce862beda&type=build ContainerLog:wss://log.bitstakehaven.com:8085/api/v1/computing/lagrange/spaces/log?space_id=201d7532-8284-4ebd-b348-ff5ce862beda&type=container NodeIdJobSourceUriSignature:0x0bca7127734cdc9c2ab2630742d7388df65b896abe5a4da4c5d8ea316234d0bb56547d7a11c627428f5afbb4205c06cfabc45acb5b80c8279ae342775723eb7b1b}" func=ReceiveJob file="cp_service.go:144"
[GIN] 2024/03/09 - 14:55:11 | 200 |  7.148365519s |   38.104.153.43 | POST     "/api/v1/computing/lagrange/jobs"
```

**Comment by ThomasBlock** [comment]
update: ubi now running

```
37106  	CPU      	fil-c2-512M	                                                                  	failed 	0.0   	2024-03-09 04:55:32	
37149  	CPU      	fil-c2-512M	                                                                  	failed 	0.0   	2024-03-09 05:25:32	
37183  	CPU      	fil-c2-512M	0x9bdd3eef469c488aa41b1d176c5932468014dc51b492d07ee8f9430039f4dd17	success	0.0   	2024-03-11 03:55:32	
37225  	CPU      	fil-c2-512M	0x6a32ec226921a05db98f710206ed3a2e96247f6a0076988ac59a759044f705a5	success	0.0   	2024-03-11 04:25:31
```

**Comment by Normalnoise** [comment]
@ThomasBlock 
> no longer appearing ( it's 15:00:00 )

we have add more obi-task to pools, CP can get the ubi task again.

**Comment by Normalnoise** [comment]
> collateral still inaccurate
HUB info for beneficiary: 8.65 sETH
HUB info for owner: 1.49 sETH
computing-provider info: Collateral(SWAN-ETH): 0.95000

@ThomasBlock 
 - The `Collateral(SWAN-ETH)` in `computing-provider info` is your CP's wallet collateral value, the wallet should be `[HUB].WalletAddress` confined in the `config.toml`

 - and the value also should be the third column in the `computing-provider collateral  info`.

 - the beneficiary address is just for receive the ubi reward.

the example is as following:

```
 computing-provider collateral  info

Address                                     Balance  Collateral  Escrow   
0x22b044AC8Ad6ce86e82CF6Aa779B401C36DEca9F  1.35345  10.00010    0.00000  
0xEF8cfb375C409312efA15d6A413C0af28cB71AF1  1.34870  0.43000     0.87000  
```
```
computing-provider info
Owner:                  0xEF8cfb375C409312efA15d6A413C0af28cB71AF1                                                                                         
Contract Address:       0x80854fb8C4BDf1A20ed73A7E2d196d6048d2a1fb                                                                                        
Multi-Address:         ------------                                                                                                  
Name:                   lag.xxxxx                                                                                                               
Node ID:                04115c33d3f7557bfd41062039d5d762aa4ae1c74b39366d3a493eaa60e567ecd1cbfcfaaf8909e9ca95b2b5d4779c9dbb8d4e39cfc7390f74ae997xxxxxx
Domain:                 lag.xxxx                                                                                                             
Running deployments:    8                                                                                                                                 
Available(SWAN-ETH):    1.34870                                                                                                                           
Collateral(SWAN-ETH):   0.43000                                                                                                                           
UBI FLAG:               Accept                                                                                                                            
Beneficiary Address:    0xEF8cfb375C409312efA15d6A413C0af28cB71AF1      
```

**Comment by Normalnoise** [comment]
> renting still problem.

I guess you have a master and a node in your K8s, so you should config the `[Registry].ServerAddress`, but the value should be `192.168.128.71:5000`, you need not to add the `http://`, and please ensure you have upgrade to [v0.4.5](https://github.com/swanchain/go-computing-provider/releases/tag/v0.4.5)

**Comment by ThomasBlock** [comment]
> > collateral still inaccurate
> > HUB info for beneficiary: 8.65 sETH
> > HUB info for owner: 1.49 sETH
> > computing-provider info: Collateral(SWAN-ETH): 0.95000
> 
> @ThomasBlock
> 
> * The `Collateral(SWAN-ETH)` in `computing-provider info` is your CP's wallet collateral value, the wallet should be `[HUB].WalletAddress` confined in the `config.toml`
> * and the value also should be the third column in the `computing-provider collateral  info`.
> * the beneficiary address is just for receive the ubi reward.
> 
> the example is as following:
> 
> ```
>  computing-provider collateral  info
> 
> Address                                     Balance  Collateral  Escrow   
> 0x22b044AC8Ad6ce86e82CF6Aa779B401C36DEca9F  1.35345  10.00010    0.00000  
> 0xEF8cfb375C409312efA15d6A413C0af28cB71AF1  1.34870  0.43000     0.87000  
> ```
> 
> ```
> computing-provider info
> Owner:                  0xEF8cfb375C409312efA15d6A413C0af28cB71AF1                                                                                         
> Contract Address:       0x80854fb8C4BDf1A20ed73A7E2d196d6048d2a1fb                                                                                        
> Multi-Address:         ------------                                                                                                  
> Name:                   lag.xxxxx                                                                                                               
> Node ID:                04115c33d3f7557bfd41062039d5d762aa4ae1c74b39366d3a493eaa60e567ecd1cbfcfaaf8909e9ca95b2b5d4779c9dbb8d4e39cfc7390f74ae997xxxxxx
> Domain:                 lag.xxxx                                                                                                             
> Running deployments:    8                                                                                                                                 
> Available(SWAN-ETH):    1.34870                                                                                                                           
> Collateral(SWAN-ETH):   0.43000                                                                                                                           
> UBI FLAG:               Accept                                                                                                                            
> Beneficiary Address:    0xEF8cfb375C409312efA15d6A413C0af28cB71AF1      
> ```
I have set [HUB].WalletAddress = beneficiary, otherwise i could not sign in to the hub

```
computing-provider info
Owner:               	0xfe017Ff8F0C7349845Ab52E58FcA96143f2c4981                                                                                         
Contract Address:    	0x8A46F39BfDFA7edc9538F27bB6D2E78597b62C0b                                                                                        	
...                                                                                                                        	
Available(SWAN-ETH): 	3.98686                                                                                                                           	
Collateral(SWAN-ETH):	0.95000                  
                                                                                                        	
computing-provider collateral info
Address                                     Balance  Collateral  Escrow   
0xfe017Ff8F0C7349845Ab52E58FcA96143f2c4981  0.16746  1.49000     0.00000  
```

**Comment by ThomasBlock** [comment]
> > renting still problem.
> 
> I guess you have a master and a node in your K8s, so you should config the `[Registry].ServerAddress`, but the value should be `192.168.128.71:5000`, you need not to add the `http://`, and please ensure you have upgrade to [v0.4.5](https://github.com/swanchain/go-computing-provider/releases/tag/v0.4.5)

okay i will try. but i think i have put http there as otherwsie it would try https but this would fail

**Comment by ThomasBlock** [comment]
> > > renting still problem.
> > 
> > 
> > I guess you have a master and a node in your K8s, so you should config the `[Registry].ServerAddress`, but the value should be `192.168.128.71:5000`, you need not to add the `http://`, and please ensure you have upgrade to [v0.4.5](https://github.com/swanchain/go-computing-provider/releases/tag/v0.4.5)
> 
> okay i will try. but i think i have put http there as otherwsie it would try https but this would fail

@Normalnoise 


with all the updates from the last days:

i can confirm that `ServerAddress = "192.168.128.71:5000" `  does not work:
`server gave HTTP response to HTTPS client`

```
Events:
  Type     Reason     Age                From               Message
  ----     ------     ----               ----               -------
  Normal   Scheduled  32s                default-scheduler  Successfully assigned ns-0x7b0cee1939a4ada062ec79f4862a42c1f47b1806/deploy-201d7532-8284-4ebd-b348-ff5ce862beda-78ccd5ffc-fbt7z to swan3
  Normal   Pulling    17s (x2 over 32s)  kubelet            Pulling image "192.168.128.71:5000/mydiffusion-ff5ce862beda:1710260883"
  Warning  Failed     17s (x2 over 32s)  kubelet            Failed to pull image "192.168.128.71:5000/mydiffusion-ff5ce862beda:1710260883": failed to pull and unpack image "192.168.128.71:5000/mydiffusion-ff5ce862beda:1710260883": failed to resolve reference "192.168.128.71:5000/mydiffusion-ff5ce862beda:1710260883": failed to do request: Head "https://192.168.128.71:5000/v2/mydiffusion-ff5ce862beda/manifests/1710260883": http: server gave HTTP response to HTTPS client
  Warning  Failed     17s (x2 over 32s)  kubelet            Error: ErrImagePull
  Normal   BackOff    5s (x3 over 32s)   kubelet            Back-off pulling image "192.168.128.71:5000/mydiffusion-ff5ce862beda:1710260883"
  Warning  Failed     5s (x3 over 32s)   kubelet            Error: ImagePullBackOff
```

and neither does  `ServerAddress = "http://192.168.128.71:5000" ` . What should i do?
`Error building Docker image: Error response from daemon: invalid reference format`

```
time="2024-03-12 17:39:33.167" level=info msg="uuid: 201d7532-8284-4ebd-b348-ff5ce862beda, spaceName: myDiffusion, hardwareName: Nvidia 4090 · 4 vCPU · 8 GiB" func=DeploySpaceTask file="cp_service.go:1051"
time="2024-03-12 17:39:33.259" level=info msg="uploading file to bucket, objectName: jobs/e7fce4e3-3b74-414a-948d-53837f439649.json, filePath: /tmp/jobs/e7fce4e3-3b74-414a-948d-53837f439649.json" func=UploadFileToBucket file="storage_service.go:52"
2024/03/12 17:39:33 Image path: build/0x7B0CEe1939a4AdA062EC79f4862a42C1F47B1806/spaces/myDiffusion
time="2024-03-12 17:39:33.697" level=error msg="Error building Docker image: Error response from daemon: invalid reference format" func=BuildImagesByDockerfile file="buildspace.go:79"
time="2024-03-12 17:39:33.697" level=info msg="Failed to extract exposed port: unable to open Dockerfile: open : no such file or directory" func=DockerfileToK8s file="deploy.go:97"
time="2024-03-12 17:39:34.403" level=error msg="http status: 400 Bad Request, code:400, url:https://api.swanipfs.com/api/v2/oss_file/get_file_by_object_name?bucket_uid=878494a8-6ab7-4694-96ac-fc89a2afcbe1&object_name=jobs/e7fce4e3-3b74-414a-948d-53837f439649.json" func=HttpRequest file="restful.go:127"
time="2024-03-12 17:39:34.404" level=error msg="https://api.swanipfs.com/api/v2/oss_file/get_file_by_object_name?bucket_uid=878494a8-6ab7-4694-96ac-fc89a2afcbe1&object_name=jobs/e7fce4e3-3b74-414a-948d-53837f439649.json failed, status:error, message:invalid param value:record not found" func=HttpRequest file="restful.go:154"
time="2024-03-12 17:39:34.404" level=error msg="https://api.swanipfs.com/api/v2/oss_file/get_file_by_object_name?bucket_uid=878494a8-6ab7-4694-96ac-fc89a2afcbe1&object_name=jobs/e7fce4e3-3b74-414a-948d-53837f439649.json failed, status:error, message:invalid param value:record not found" func=HttpGet file="restful.go:64"
[GIN] 2024/03/12 - 17:39:34 | 200 |      67.839µs |   38.104.153.43 | GET      "/api/v1/computing/host/info"
```

**Comment by Normalnoise** [comment]
> > > collateral still inaccurate
> > > HUB info for beneficiary: 8.65 sETH
> > > HUB info for owner: 1.49 sETH
> > > computing-provider info: Collateral(SWAN-ETH): 0.95000
> > 
> > 
> > @ThomasBlock
> > 
> > * The `Collateral(SWAN-ETH)` in `computing-provider info` is your CP's wallet collateral value, the wallet should be `[HUB].WalletAddress` confined in the `config.toml`
> > * and the value also should be the third column in the `computing-provider collateral  info`.
> > * the beneficiary address is just for receive the ubi reward.
> > 
> > the example is as following:
> > ```
> >  computing-provider collateral  info
> > 
> > Address                                     Balance  Collateral  Escrow   
> > 0x22b044AC8Ad6ce86e82CF6Aa779B401C36DEca9F  1.35345  10.00010    0.00000  
> > 0xEF8cfb375C409312efA15d6A413C0af28cB71AF1  1.34870  0.43000     0.87000  
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > computing-provider info
> > Owner:                  0xEF8cfb375C409312efA15d6A413C0af28cB71AF1                                                                                         
> > Contract Address:       0x80854fb8C4BDf1A20ed73A7E2d196d6048d2a1fb                                                                                        
> > Multi-Address:         ------------                                                                                                  
> > Name:                   lag.xxxxx                                                                                                               
> > Node ID:                04115c33d3f7557bfd41062039d5d762aa4ae1c74b39366d3a493eaa60e567ecd1cbfcfaaf8909e9ca95b2b5d4779c9dbb8d4e39cfc7390f74ae997xxxxxx
> > Domain:                 lag.xxxx                                                                                                             
> > Running deployments:    8                                                                                                                                 
> > Available(SWAN-ETH):    1.34870                                                                                                                           
> > Collateral(SWAN-ETH):   0.43000                                                                                                                           
> > UBI FLAG:               Accept                                                                                                                            
> > Beneficiary Address:    0xEF8cfb375C409312efA15d6A413C0af28cB71AF1      
> > ```
> 
> I have set [HUB].WalletAddress = beneficiary, otherwise i could not sign in to the hub
> 
> ```
> computing-provider info
> Owner:               	0xfe017Ff8F0C7349845Ab52E58FcA96143f2c4981                                                                                         
> Contract Address:    	0x8A46F39BfDFA7edc9538F27bB6D2E78597b62C0b                                                                                        	
> ...                                                                                                                        	
> Available(SWAN-ETH): 	3.98686                                                                                                                           	
> Collateral(SWAN-ETH):	0.95000                  
>                                                                                                         	
> computing-provider collateral info
> Address                                     Balance  Collateral  Escrow   
> 0xfe017Ff8F0C7349845Ab52E58FcA96143f2c4981  0.16746  1.49000     0.00000  
> ```

so have you config the [HUB].WalletAddress="0xfe017Ff8F0C7349845Ab52E58FcA96143f2c4981"
please give your config

**Comment by ThomasBlock** [comment]
> > > > collateral still inaccurate
> > > > HUB info for beneficiary: 8.65 sETH
> > > > HUB info for owner: 1.49 sETH
> > > > computing-provider info: Collateral(SWAN-ETH): 0.95000
> > > 
> > > 
> > > @ThomasBlock
> > > 
> > > * The `Collateral(SWAN-ETH)` in `computing-provider info` is your CP's wallet collateral value, the wallet should be `[HUB].WalletAddress` confined in the `config.toml`
> > > * and the value also should be the third column in the `computing-provider collateral  info`.
> > > * the beneficiary address is just for receive the ubi reward.
> > > 
> > > the example is as following:
> > > ```
> > >  computing-provider collateral  info
> > > 
> > > Address                                     Balance  Collateral  Escrow   
> > > 0x22b044AC8Ad6ce86e82CF6Aa779B401C36DEca9F  1.35345  10.00010    0.00000  
> > > 0xEF8cfb375C409312efA15d6A413C0af28cB71AF1  1.34870  0.43000     0.87000  
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > ```
> > > computing-provider info
> > > Owner:                  0xEF8cfb375C409312efA15d6A413C0af28cB71AF1                                                                                         
> > > Contract Address:       0x80854fb8C4BDf1A20ed73A7E2d196d6048d2a1fb                                                                                        
> > > Multi-Address:         ------------                                                                                                  
> > > Name:                   lag.xxxxx                                                                                                               
> > > Node ID:                04115c33d3f7557bfd41062039d5d762aa4ae1c74b39366d3a493eaa60e567ecd1cbfcfaaf8909e9ca95b2b5d4779c9dbb8d4e39cfc7390f74ae997xxxxxx
> > > Domain:                 lag.xxxx                                                                                                             
> > > Running deployments:    8                                                                                                                                 
> > > Available(SWAN-ETH):    1.34870                                                                                                                           
> > > Collateral(SWAN-ETH):   0.43000                                                                                                                           
> > > UBI FLAG:               Accept                                                                                                                            
> > > Beneficiary Address:    0xEF8cfb375C409312efA15d6A413C0af28cB71AF1      
> > > ```
> > 
> > 
> > I have set [HUB].WalletAddress = beneficiary, otherwise i could not sign in to the hub
> > ```
> > computing-provider info
> > Owner:               	0xfe017Ff8F0C7349845Ab52E58FcA96143f2c4981                                                                                         
> > Contract Address:    	0x8A46F39BfDFA7edc9538F27bB6D2E78597b62C0b                                                                                        	
> > ...                                                                                                                        	
> > Available(SWAN-ETH): 	3.98686                                                                                                                           	
> > Collateral(SWAN-ETH):	0.95000                  
> >                                                                                                         	
> > computing-provider collateral info
> > Address                                     Balance  Collateral  Escrow   
> > 0xfe017Ff8F0C7349845Ab52E58FcA96143f2c4981  0.16746  1.49000     0.00000  
> > ```
> 
> so have you config the [HUB].WalletAddress="0xfe017Ff8F0C7349845Ab52E58FcA96143f2c4981" please give your config

Thank you for the feedback. But i told you about the "i can not login on the hub with the other address" problem 2 months ago and you chose not to reply. this far in the testnet i will not change my hub adress as i expect to change my testnet score!?

can we please focus on the bigger problems?

1. Ubi taks work, but they do no longer pay out
2. We can only deploy simple tasks like pacman and no real GPU tasks like Stable Diffusion

Here is my config. How can we get [Registry].ServerAddress correct? ( https://github.com/swanchain/go-computing-provider/issues/40#issuecomment-1992099738 )

```
[API]
Port = 8085                                     # The port number that the web server listens on
MultiAddress = "/ip4/X.Y.X.Y/tcp/8085"    # The multiAddress for libp2p
Domain = "xyz.com"                                     # The domain name
NodeName = "ThomasBlock.io-Test"                                   # The computing-provider node name
UbiTask = true
RedisUrl = "redis://127.0.0.1:6379"           # The redis server address
RedisPassword = ""                            # The redis server access password

[UBI]
UbiTask = true                                              # Accept the UBI task (Default: true)
UbiEnginePk = "0xB5aeb540B4895cd024c1625E146684940A849ED9"  # UBI Engine's public key, CP only accept the task from this UBI engine
UbiUrl ="https://ubi-task.swanchain.io/v1"                   # UBI Engine's API address

[LOG]
CrtFile = "/home/user/ssl/fullchain-wild.pem"   # Your domain name SSL .crt file path
KeyFile = "/home/user/ssl/privkey-wild.pem"   # Your domain name SSL .key file path

[HUB]
ServerUrl = "https://orchestrator-api.swanchain.io" # The Orchestrator's API address
AccessToken = "XYZ"
WalletAddress = "0x269EBeee083CE6f70486a67dC8036A889bF322A9"
BalanceThreshold= 0.4                                # The cp’s collateral balance threshold

[MCS]
ApiKey = "xyz"                                   # Acquired from "https://www.multichain.storage" -> setting -> Create API Key
BucketName = "B1"                               # Acquired from "https://www.multichain.storage" -> bucket -> Add Bucket
Network = "polygon.mumbai"                   # polygon.mainnet for mainnet, polygon.mumbai for testnet
FileCachePath = "/tmp"                        # Cache directory of job data

[Registry]
ServerAddress = "http://192.168.128.71:5000"                            # The docker container image registry address, if only a single node, you can ignore
UserName = ""                                 # The login username, if only a single node, you can ignore
Password = ""                                 # The login password, if only a single node, you can ignore

[RPC]
SWAN_TESTNET ="https://saturn-rpc.swanchain.io"     # Swan testnet RPC
SWAN_MAINNET= ""                                    # Swan mainnet RPC

[CONTRACT]
SWAN_CONTRACT="0x91B25A65b295F0405552A4bbB77879ab5e38166c"              # Swan token's contract address
SWAN_COLLATERAL_CONTRACT="0xdc200f89258e72aC3602dD23BD3642C4bd4eE34e"   # Swan's collateral address
```

**Comment by Normalnoise** [comment]
Let's start test v0.4.6
